services/requester/requester.go

```diff
@@ -152,6 +153,54 @@ func (e *EVM) SendRawTransaction(ctx context.Context, data []byte) (common.Hash,
 		return common.Hash{}, err
 	}
 
+	txDataLen := len(tx.Data())
+
+	// Contract creation doesn't validate call data, handle first
+	if tx.To() == nil {
+		// Contract creation should contain sufficient data to deploy a contract. A
+		// typical error is omitting sender due to some quirk in the javascript call
+		// e.g. https://github.com/onflow/go-ethereum/issues/16106.
+		if txDataLen == 0 {
+			// Prevent sending ether into black hole (show stopper)
+			if tx.Value().Cmp(big.NewInt(0)) > 0 {
+				return common.Hash{}, errors.New("transaction will create a contract with value but empty code")
+			}
+			// No value submitted at least, critically Warn, but don't blow up
+			return common.Hash{}, errors.New("transaction will create a contract with empty code")
+		} else if txDataLen < 40 { // arbitrary heuristic limit
```

Comment:
no need for else since above if will always return, 
also where did we get this limit from? is this something they do in go-eth? it seems weird

Reply:
Good catch :+1: 
Updated in https://github.com/onflow/flow-evm-gateway/pull/314/commits/2b2a4cca9d5a2b3832c4f61b636fca6aea786f89
That limit is indeed taken from `geth`. I guess the smallest possible Solidity contract, does not produce a bytecode less than `40` bytes.